### PR TITLE
drivers/spi: Add support for GRLIB SPIMCTRL

### PR DESCRIPTION
**Pull request by tbr-tt**
This adds support for the GRLIB SPIMCTRL SPI controller used in LEON and NOEL-V systems. SPIMCTRL can operate in two different modes: In the default mode it allows memory-mapped read access to the flash data. When set in the user mode, it can be used to generate SPI bus transactions.

Also enables SPIMCTRL support on the board gr716a_mini.

Tested successfully on gr716a_mini with:
- `samples/drivers/spi_flash`
- `samples/drivers/flash_shell`

### CODE REVIEW

File: drivers/spi/spi_grlib_spimctrl.c [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+	ctx->config = config;
+
+	return 0;
+}
+
+static int transceive(const struct device *dev,
+				  const struct spi_config *config,
+				  const struct spi_buf_set *tx_bufs,
+				  const struct spi_buf_set *rx_bufs)
```
Comment (tbursztyka) [inline drivers/spi/spi_grlib_spimctrl.c:100]: weird indentation, try to configure your editor to follow linux code style.
Basically, parameters should indent below the ( + 1 space

File: dts/bindings/spi/gaisler,spimctrl.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+compatible: "gaisler,spimctrl"
+
+include: spi-controller.yaml
+
+properties:
+  reg:
+    required: true
+
+  mem:
```
Comment (tbursztyka) [inline dts/bindings/spi/gaisler,spimctrl.yaml:14]: I could not find where this is used in the driver

**Review by tbursztyka — COMMENTED** [review]
Just a style issue, rest lgtm

File: drivers/spi/spi_grlib_spimctrl.c (hunk 1 above)
Comment (tbr-tt) [reply drivers/spi/spi_grlib_spimctrl.c]: Thanks. The indentation has now been updated for the functions `transceive`, `transceive_async` and `release`.

File: dts/bindings/spi/gaisler,spimctrl.yaml (hunk 2 above)
Comment (tbr-tt) [reply dts/bindings/spi/gaisler,spimctrl.yaml]: It was not used and is now removed from the .yaml file.

**Review by tbursztyka — APPROVED** [review]

**Review by julius-barendt — APPROVED** [review]
LGTM
Also tested on hardware with my GR716a mini

**Pull request merged**